### ISSUE TITLE
Tables migration guide - guidance on migrating segmented queries, with continuation tokens

### ISSUE THREAD

**Issue by Bellarmine-Head** [Azure/azure-sdk-for-net#28414]
### Library name and version

Azure.Data.Tables @ 12.5.0

### Query/Question

I have CloudTable code that calls `ExecuteQuerySegmented` to get (say) 200 entities at a time.  It passes back the entities, plus the returned continuation token (base64-encoded), from the web server to the web client.  When the user clicks on "Load more" the web client makes another request, and passes up the continuation token.  Pretty easy stuff.

I am working my way through translating this into `Azure.Data.Tables`.  It's not plain sailing.

Perhaps you can give some guidance in this ticket, and also include this topic in the Migration Guide?

### Environment

Windows 10
.NET 6
Visual Studio 2022

**Comment by Bellarmine-Head** [comment]
I'm looking for a way of passing in a continuation token (sent up from the web client to the web server) into _subsequent_ calls to https://docs.microsoft.com/en-us/dotnet/api/azure.data.tables.tableclient.query?view=azure-dotnet but I'm not seeing it.

I don't want to have to cache `Azure.Pageable<MyTableEntity> queryResults` in web server memory (e.g. session state) for use next time (and there may not be a next time, if the user doesn't click on "Load more").

Not sure what to do here....

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by Bellarmine-Head** [comment]
This is providing some clues....

https://github.com/Tazmainiandevil/TableStorage.Abstractions/blob/master/src/TableStorage.Abstractions/Store/TableStore.cs

What I'm starting to (belatedly) realize is that maybe the call to https://docs.microsoft.com/en-us/dotnet/api/azure.data.tables.tableclient.query?view=azure-dotnet doesn't do the query... only iterating through the reults does (as per LINQ's lazy evaluation).

**Comment by christothes** [comment]
> This is providing some clues....
> 
> https://github.com/Tazmainiandevil/TableStorage.Abstractions/blob/master/src/TableStorage.Abstractions/Store/TableStore.cs
> 
> What I'm starting to (belatedly) realize is that maybe the call to https://docs.microsoft.com/en-us/dotnet/api/azure.data.tables.tableclient.query?view=azure-dotnet doesn't execute the query... only iterating through the results does (as per LINQ's lazy evaluation).

This is correct - calling `Query` doesn't immediately execute the query, it generates the enumerable. Iterating through the enumerable executes queries to the service as needed to fill the next page of results.

**Comment by Bellarmine-Head** [comment]
@christothes 

Ok, this is working....

```csharp
/* Decode/deserialize the next-page continuation token, if specified. */

String tableContinuationToken = null;
if (!String.IsNullOrWhiteSpace(continuationToken))
{
    try
    {
        tableContinuationToken = DeserializeContinuationToken(continuationToken);
    }
    catch (Exception ex)
    {
        throw new InternalServiceFailureException(
            $"Failed to deserialize the next-page continuation token.  {ex.Message}");
    }
}


/* Make the query, and return the results. */

try
{
    var query = tableClient.Query<AwsSnsNotificationMessageTableEntity>(
        filter: e => e.PartitionKey == organizationId.ToString(),
        maxPerPage: MaxNumEmailsPerPage);

    var pagedQuery = query.AsPages(tableContinuationToken, pageSizeHint: MaxNumEmailsPerPage);

    var firstOrNextPage = pagedQuery.FirstOrDefault();

    var summaries = firstOrNextPage?.Values?
        .Select(EntityToEmailSummary)
        .Where(summary => summary != null)
        .ToArray() ?? Array.Empty<TrainingCampaignEmailSummaryModel>();

    continuationToken = firstOrNextPage?.ContinuationToken == null ? null :
        SerializeContinuationToken(firstOrNextPage.ContinuationToken);

    return new TrainingCampaignEmailSummariesPageModel
    {
        TenantCode = tenantCode,
        OrganizationId = organizationId,
        TrainingCampaignEmailsNotBeingLoggedForTenant = false,
        NextPageContinuationToken = continuationToken,
        Summaries = summaries
    };
}
catch (Exception ex)
{
    throw new InternalServiceFailureException(
        $"Failed to query the {StorageTableName} table for training campaign emails.  {ex.Message}");
}
```

Notes:-

1. I am serializing and deserializing the continuation token to ensure that it is base-64 encoded in a URL-safe manner (since its format isn't documented, I don't think).
2. The iteration code is within the `try { ... }` deliberately, whereas before I only had the call to `TableClient.Query` call in the try.
3. Page size is specified wherever possible, just to be sure.
4. Testing out the code with tiny page size (2) works nicely, since I don't have a large dataset to hand on my dev env.
